public/apps/account/password-reset-panel.tsx

```diff
@@ -0,0 +1,157 @@
+import { InternalUserUpdate } from '../configuration/types';
+import { buildUrl } from '../configuration/utils/url-builder';
+
+interface PasswordResetPanelProps extends AppDependencies {
+  username: string;
+  handleClose: () => void;
+}
+
+export function PasswordResetPanel(props: PasswordResetPanelProps) {
+  const [currentPassword, setCurrentPassword] = useState<string>('');
+  // reply on backend response of login call to verify
+  const [isCurrentPasswordInvalid, setIsCurrentPasswordInvalid] = useState<boolean>(false);
+
+  const [newPassword, setNewPassword] = useState<string>('');
+  // reply on backend response of user update call to verify
+  const [isNewPasswordInvalid, setIsNewPasswordInvalid] = useState<boolean>(false);
+
+  const [repeatNewPassword, setRepeatNewPassword] = useState<string>('');
+  const [isRepeatNewPasswordInvalid, setIsRepeatNewPasswordInvalid] = useState<boolean>(false);
+
+  const handleReset = async () => {
+    // validate the current password
+    try {
+      await props.coreStart.http.post('/auth/login', {
+        body: JSON.stringify({
+          username: props.username,
+          password: currentPassword,
+        }),
+      });
+    } catch (e) {
+      setIsCurrentPasswordInvalid(true);
+      return;
+    }
+
+    // update new password
+    try {
+      const user = await getUserDetail(props.coreStart.http, props.username);
+
+      const updateObject: InternalUserUpdate = {
+        password: newPassword,
+        backend_roles: user.backend_roles,
+        attributes: user.attributes,
+      };
+
+      await updateUser(props.coreStart.http, props.username, updateObject);
+
+      // need to log in with new password
+      window.location.href = buildUrl();
+    } catch (e) {
+      console.error(
+        'Password reset failed, please check console log for more detail. ' + JSON.stringify(e)
+      );
+      setIsNewPasswordInvalid(true);
+    }
+  };
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={props.handleClose}>
+        <EuiSpacer />
+        <EuiModalBody>
+          <EuiTitle>
+            <h4>Reset password for &quot;{props.username}&quot;</h4>
+          </EuiTitle>
+
+          <EuiSpacer />
+
+          <FormRow
+            headerText="Current password"
+            helpText="Verify your account by entering your current password"
+          >
+            <EuiFieldText
+              prepend={<EuiIcon type="lock" />}
+              type="password"
+              onChange={function (e: React.ChangeEvent<HTMLInputElement>) {
+                setCurrentPassword(e.target.value);
+                setIsCurrentPasswordInvalid(false);
+              }}
+              isInvalid={isCurrentPasswordInvalid}
+            />
+          </FormRow>
+
+          <FormRow
+            headerText="New password"
+            helpText="The password must contain from m to n characters. Valid characters are [for example lowercase a-z
+            , 0-9, and - (hyphen)]"
```

Comment:
I don't think backend limit the characters set to this, this is for password rather than a name, special chars like `!@#$%^&*()` should work. Let's confirm with UX designer for the help text

Reply:
To unblock, for now, I will leave a TODO to update the text after UX provides it.